api/request_source/request_source_plugin.proto

```diff
@@ -0,0 +1,35 @@
+// Config protos for the Request Source Plugin Config Factories.
+syntax = "proto3";
+
+package nighthawk.request_source;
+
+import "google/protobuf/wrappers.proto";
+import "validate/validate.proto";
+import "api/client/options.proto";
+
+// Configuration for FileBasedPluginRequestSource (plugin name:
+// "nighthawk.file-based-request-source-plugin")
+// The factory will load the RequestOptionsList from the file, and then passes it to the
+// requestSource it generates. The resulting request source will loop over the RequestOptionsList it
+// is passed.
+message FileBasedPluginConfig {
+  // The file_path is the path to a file that contains a RequestOptionList in json or yaml format.
+  string file_path = 1;
+  // The pluginfactory makes requestSources that will generate requests from the RequestOptionList
+  // up to num_requests number of times. If num_requests exceeds the number of RequestOptions in the
+  // RequestOptionList located in the file at file_path, it will loop. num_requests = 0 means no
+  // limit on the number of requests to be produced.
+  google.protobuf.UInt32Value num_requests = 2 [(validate.rules).uint32 = {gte: 0, lte: 1000000}];
+  // The pluginfactory will load the file located in file_path as long as it is below max_file_size,
+  // if it's too large it will throw an error.
+  google.protobuf.UInt32Value max_file_size = 3 [(validate.rules).uint32 = {lte: 1000000}];
+}
+
+// Configuration for StubPluginRequestSource (plugin name: "nighthawk.stub-request-source-plugin")
+// The plugin does nothing. This is for testing and comparison of the Request Source Plugin Factory
+// mechanism using a minimal version of plugin that does not require a more complicated proto or
+// file reading.
+message StubPluginConfig {
+  // test input value.
```

Comment:
We had a comment thread about improving this comment, but looks like the it wasn't actually done. Repeating the comment:


Can we improve this comment? We could say what the input argument is for / what does the plugin do with it. It raises a question as to why a plugin that does nothing need an input argument.

Reply:
Oh I thought I fixed it by making it do something.